modules/porous_flow/src/dirackernels/PorousFlowPointSourceFromPostprocessor.C

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPointSourceFromPostprocessor.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPointSourceFromPostprocessor);
+
+InputParameters
+PorousFlowPointSourceFromPostprocessor::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+  params.addRequiredParam<PostprocessorName>(
+      "mass_flux",
+      "The postprocessr name holding the mass flux at this point in kg/s (positive is flux in, "
+      "negative is flux out)");
+  params.addRequiredParam<Point>("point", "The x,y,z coordinates of the point source (sink)");
```

Comment:
I think i prefer "(or sink)" instead of the more terse "(sink)".   I'll leave it up to you.